PharmacyApi/Services/IPharmacyService.cs

```diff
@@ -1,11 +1,12 @@
 ﻿using PharmacyApi.Models;
+using PharmacyApi.Utilities;
 
 namespace PharmacyApi.Services;
 
 public interface IPharmacyService
 {
-    IEnumerable<Pharmacy> GetAll();
-    Task<Pharmacy?> GetById(int id);
-    Task<Pharmacy?> UpdateById(int id, Pharmacy updatedPharmacy);
+    Task<ServiceResult<IAsyncEnumerable<Pharmacy>>> GetPharmacyListAsync();
+    Task<ServiceResult<Pharmacy>> GetByIdAsync(int id);
+    Task<ServiceResult<Pharmacy>> UpdateByIdAsync(Pharmacy updatedPharmacy);
```

Comment:
Dont name methods with generic names - unless you are supplying a truly generic function - which this is definitely not...
Use the Pharmacy in the name all the way up and down.